ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionProcessor.java

```diff
@@ -328,7 +334,14 @@ public TransactionProcessingResult processTransaction(
               transaction.getPayload(), transaction.isContractCreation());
       final long accessListGas =
           gasCalculator.accessListGasCost(accessListEntries.size(), accessListStorageCount);
-      final long gasAvailable = transaction.getGasLimit() - intrinsicGas - accessListGas;
+      final long dataGas;
+      if (feeMarket.implementsDataFee()) {
+        dataGas = gasCalculator.dataGasCost(transaction.getVersionedHashes().orElseThrow().size());
```

Comment:
we can use blobcount for this one or remove this one because we are already calculating this value in 306 line

Reply:
good catch, fixed